src/main/java/seedu/address/model/task/Task.java

```diff
@@ -0,0 +1,91 @@
+package seedu.address.model.task;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents a Task that can be assigned to a room.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Task {
+
+    // Data fields (i.e. values entered by user).
+    private final Description description;
+    private final DateTimeDue dueAt;
+
+    // Metadata (i.e. values generated upon creation).
+    private final DateTimeCreated createdAt;
```

Comment:
I was just wondering why you need to have a date time created attribute for task :P

Reply:
i think he used it because it allows easier tracking of the tasks due date using DateTime instead of a string

Reply:
As in I meant the attribute "DateTimeCreated" haha, like why do you need to know when the task was created :P . The tracking of tasks due date is by attribute "DateTimeDue".

Reply:
Actually, yeah. It is probably not needed for an MVP.

The rationale behind adding `DateTimeCreated` is to allow the user to filter tasks with creation date as a criterion. For example, in [Slack](https://slack.com/intl/en-sg/solutions/task-management), users can see when the task is created. It is helpful when the user wants to query what tasks were created yesterday or within the last few days.

Right now, the plan is for the user to click on a `Room` in order to view all tasks assigned to it. There is no separate tab for the users to manage tasks. In the future, when we implement reminders or a search for tasks, `DateTimeCreated` might be more useful.

I think I will remove this from the MVP. Unit testing for this class also requires more work because we would need another layer of abstraction to handle and control the date-time instead of using `LocalDateTime.now()`.

Reply:
Sounds good!

Reply:
Okay!